bolt/test/runtime/AArch64/hugify.c

```diff
@@ -0,0 +1,34 @@
+// Make sure BOLT correctly processes --hugify option
+
+#include <stdio.h>
+
+int g1 = 1;
+int g2;
+static int sg1 = 1;
+static int sg2;
+
+
+int main(int argc, char **argv) {
+  printf("Hello world %p = %d , %p = %d\n", &g1, g1, &sg1, sg1);
+  printf("%p = %d , %p = %d\n", &g2, g2, &sg2, sg2);
+  return 0;
+}
+
+/*
+REQUIRES: system-linux,bolt-runtime
+
+RUN: %clang %cflags -no-pie %s -o %t.nopie.exe -Wl,-q
```

Comment:
probably we need to exclude %cflags, because lit.local.cfg contains -fPIE option by defualt.

Reply:
Hi, @yavtuk. 

Thanks for highlighting this. 

Just a minor addition. The hugify test for AArch64 was slightly modified. That is why initially I kept it separated.

Regarding %cflags, If remove it what we should do with the following:

`--target=aarch64-unknown-linux-gnu -fPIE -fuse-ld=lld -Wl,--unresolved-symbols=ignore-all -pie`
 
maybe we need to exclude -fPIE you mean for no-pie test? 

Though, few tests were added as well: 
- section-order.test
- user-func-reorder.c

Personally, I would prefer to have required flags specified explicitly in test file.  

If you have some ideas about other tests that could benefit AArch64 from X86 set, let me know, I will try to check them.

Reply:
ok, let’s leave separate tests, I will look at x86

Reply:
@yavtuk 
btw, did  I address your comment below: 
>  we can move bolt/test/runtime/AArch64/hugify.c -> bolt/test/runtime/hugify.c and use the test for both platforms

And, you agree to avoid removing %cflags to keep substitutions by lit as it is? ( we maybe can handle duplicates in other tests just fixing configs if there are some worries about it.

Reply:
> @yavtuk 
> btw, did  I address your comment below: 
> >  we can move bolt/test/runtime/AArch64/hugify.c -> bolt/test/runtime/hugify.c and use the test for both platforms
> 
> And, you agree to avoid removing %cflags to keep substitutions by lit as it is? ( we maybe can handle duplicates in other tests just fixing configs if there are some worries about it. 

Yes, you did, for x86 we need to delete flags for no-pie test, I will check it locally. Now I agree with your solution. Thanks for that

Reply:
We expect to have %cflags in every test, it is better to avoid situiations when %cflags is excluded

Reply:
> We expect to have %cflags in every test, it is better to avoid situiations when %cflags is excluded

Ok, I will prepare a patch and share it with you